js/maps.js

```diff
@@ -220,13 +209,32 @@ function zoomToFit() {
     });
   map.fitBounds(bounds);
   map.panToBounds(bounds);
-  
+
 }
 
 MAP_SELECT.addEventListener('change', toggleLayer);
 MAP_THRESHOLD.addEventListener('change', renderMap);
 ZOOM_TO_FIT.addEventListener('click', zoomToFit);
 
-API[getLayers](eventName)
+const center = new google.maps.LatLng(15.231458142, -61.2507115);
+const map = new google.maps.Map(MAP_CONTAINER, Object.assign(MAP_OPTIONS, { center }));
+
+const eventName = queryParams().event;
+const pendingLayers = queryParams().pending;
+let getLayerFunc = 'layers';
+let layer = queryParams().layer;
+let zoomFunc = FitEventBounds;
+// pending query param overrides the layer query param
+// layer is used to overide the map to only show
+// a single layer source, e.g. for snapshots
+if (pendingLayers) {
+  getLayerFunc = 'pendingLayers';
+} else if (layer) {
+  getLayerFunc = 'layer';
+  zoomFunc = zoomToFit;
+}
+
+API[getLayerFunc](eventName, layer)
   .then(buildLayersMenu)
-  .then(renderMapAndFit);
+  .then(renderMap)
+  .then(zoomFunc);
```

Comment:
What you want to do here, I think, is pass `zoomFunc` to Papaparse as the `complete` callback in the parser config. https://www.papaparse.com/docs#config

Maybe pass it as an argument to the `readMap` function?

Reply:
I've done it slightly differently and promisified the `renderMap()` function and used the promise resolve as the callback to the parser config. 

I'm not sure if passing the callback function to renderMap is more optimal and clearer / cleaner in the code? Happy to change.

Reply:
Sounds good to me. `renderMap()` is set up to handle an array of layers, so you might need to use `Promise.all()`. I can never remember how iteration works for `nodeLists` returned from query selectors, which are slightly different from arrays.